pkg/install/resources.go

```diff
@@ -0,0 +1,161 @@
+package install
+
+import (
+	"fmt"
+
+	appsv1beta1 "k8s.io/api/apps/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1beta1 "k8s.io/api/rbac/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type Option func(*config)
+
+type config struct {
+	container string
+	tag       string
+}
+
+func (c *config) image() string {
+	return fmt.Sprintf("%s:%s", c.container, c.tag)
+}
+
+func WithContainer(container string) Option {
+	return func(c *config) {
+		c.container = container
+	}
+}
+
+func WithTag(tag string) Option {
+	return func(c *config) {
+		c.tag = tag
+	}
+}
+
+func labels() map[string]string {
+	return map[string]string{
+		"component": "ark",
+	}
+}
+
+func objectMeta(name, namespace string) metav1.ObjectMeta {
```

Comment:
I typically think in terms of `namespace, name` - would you consider swapping the order?